test/testcafe/spec/EnrollAuthenticatorEmail_spec.js

```diff
@@ -146,6 +335,96 @@ test
     await t.expect(lastRequestUrl).eql('http://localhost:3000/idp/idx/challenge/resend');
   });
 
+test
+  .requestHooks(logger, validOTPmockWithEmailMagicLink)('resend after 30 seconds with EML', async t => {
+    const enrollEmailPageObject = await setup(t);
+    await t.expect(enrollEmailPageObject.resendEmail.isHidden()).ok();
+    await t.wait(31000);
+    await t.expect(enrollEmailPageObject.resendEmail.isHidden()).notOk();
+    await t.expect(enrollEmailPageObject.resendEmail.getText()).eql('Haven\'t received an email? Send again');
+
+    // 8 poll requests in 31 seconds and 1 resend request after click.
+    await t.expect(logger.count(
+      record => record.response.statusCode === 200 &&
+        record.request.url.match(/poll/)
+    )).eql(8);
+
+    await enrollEmailPageObject.resendEmail.click();
+
+    await t.expect(enrollEmailPageObject.resendEmail.isHidden()).ok();
+    await t.expect(logger.count(
+      record => record.response.statusCode === 200 &&
+      record.request.url.match(/resend/)
+    )).eql(1);
+
+    const { request: {
+      body: firstRequestBody,
+      method: firstRequestMethod,
+      url: firstRequestUrl,
+    }
+    } = logger.requests[0];
+    const { request: {
+      body: lastRequestBody,
+      method: lastRequestMethod,
+      url: lastRequestUrl,
+    }
+    } = logger.requests[logger.requests.length - 1];
+    let jsonBody = JSON.parse(firstRequestBody);
+    await t.expect(jsonBody).eql({'stateHandle':'eyJ6aXAiOiJER'});
+    await t.expect(firstRequestMethod).eql('post');
+    await t.expect(firstRequestUrl).eql('http://localhost:3000/idp/idx/challenge/poll');
+
+    jsonBody = JSON.parse(lastRequestBody);
+    await t.expect(jsonBody).eql({'stateHandle':'eyJ6aXAiOiJER'});
+    await t.expect(lastRequestMethod).eql('post');
+    await t.expect(lastRequestUrl).eql('http://localhost:3000/idp/idx/challenge/resend');
+  });
+
+test
+  .requestHooks(logger, validOTPmockWithoutEmailMagicLink)('resend after 30 seconds without EML', async t => {
+    const enrollEmailPageObject = await setup(t);
+    await t.expect(enrollEmailPageObject.resendEmail.isHidden()).ok();
+    await t.wait(31000);
```

Comment:
Do you know if we're allowed to use explicit `wait`s in testcafe tests?

Reply:
Yes and this is similar to ChallengeAuthenticatorView where the test scenario is for resend. Also, I have discussed this with Aaron on exploring a way to configure this timeout for testcafe.